finesse/hardware/plugins/time/ntp_time.py

```diff
@@ -49,23 +54,33 @@ def __init__(
         self._ntp_port = ntp_port
         self._ntp_timeout = ntp_timeout
 
-    def close(self) -> None:
-        """Close the connection to the device."""
-
-    def get_time(self) -> float:
-        """Get the current time.
+        # Set up time offset polling.
+        self.poll_time_offset()
+        self._poll_timer = QTimer()
+        self._poll_timer.timeout.connect(self.poll_time_offset)
+        self._poll_timer.start(int(ntp_poll_interval * 1000))
 
-        Returns:
-            The current time in seconds since the epoch.
-        """
+    def poll_time_offset(self) -> None:
+        """Query the NTP server for the current time offset."""
```

Comment:
This looks good, but I feel like this function could do with some logging. It would be useful to know when the NTP server is being polled and, if successful, what the offset is. Given that we won't be polling too often, this shouldn't create much log spam.